state/factory/workingset.go

```diff
@@ -495,7 +496,18 @@ func (ws *workingSet) pickAndRunActions(
 			if !ok {
 				break
 			}
-			if nextAction.GasLimit() > blkCtx.GasLimit {
+
+			// IntrinsicGas is used for all non-execution actions, and GasLimit is used for execution actions
+			var gaslimit uint64 = math.MaxUint64
+			if _, ok := nextAction.Action().(*action.Execution); ok {
+				gaslimit = nextAction.GasLimit()
+			} else {
+				gaslimit, err = nextAction.IntrinsicGas()
```

Comment:
I don't mind dropping this action as well as the actions of the account

Reply:
this change will need a hard-fork?